aws-datastore/src/main/java/com/amplifyframework/datastore/DataStoreConfiguration.java

```diff
@@ -305,6 +339,23 @@ public Builder syncPageSize(@IntRange(from = 0) Integer syncPageSize) {
             return Builder.this;
         }
 
+        /**
+         * Sets a sync expression for a particular model to filter which data is synced locally.  The expression
+         * is evaluated each time DataStore is started.  The QueryPredicate is applied on both sync and subscriptions.
```

Comment:
From the PR description:
> Runtime Modification: Clients can force a re-evaluation of their DataStoreSyncExpression at runtime by calling DataStore.stop(), DataStore.start(), if they wish to modify the filter to obtain a different set of data.

However, if I need to provide the sync expressions before calling `Amplify.configure(...)`, how could I change the expressions later? (Perhaps the sync expression itself has to be dynamic?)

Reply:
Here's how I'm doing it in my sample app:

```
AmplifyFactory.setBlogQueryPredicate(Blog.NAME.contains("a"));
Amplify.DataStore.stop(...);
Amplify.DataStore.start(...);

public class AmplifyFactory {
    private static QueryPredicate blogQueryPredicate = QueryPredicates.all();

    public static void setBlogQueryPredicate(QueryPredicate queryPredicate) {
        blogQueryPredicate = queryPredicate;
    }

    public static void configure(Context context) {
        try {
            Amplify.addPlugin(new AWSDataStorePlugin(DataStoreConfiguration.builder()
                    .syncExpression(Blog.class, () -> blogQueryPredicate)
                    .build()));
            Amplify.configure(context);
        } catch (AmplifyException exception){
            LOG.error(exception.getMessage(), exception);
        }
    }
}
```

Reply:
Seems ~reasonable. Or someone really wanted, they could make like a:

```java
final class MutableSyncExpression implements DataStoreSyncExpression {
    private QueryPredicate queryPredicate;

    MutableSyncExpression() {
        this.queryPredicate = QueryPredicates.all();
    }

    @Override public QueryPredicate resolvePredicate() {
        return queryPredicate;
    }

    void updatePredicate(QueryPredicate queryPredicate) {
        this.queryPredicate = queryPredicate;
    }
    // equals/hashCode/toString ...
}
```